### ISSUE TITLE
Redisson script caching

### ISSUE THREAD

**Issue by alturkovic** [redisson/redisson#1676]
According to the [Redis documentation](https://redis.io/commands/eval#bandwidth-and-evalsha), script caching is recommended, using `EVALSHA `and `EVAL `only if Redis server does not have the script already loaded, but I don't see this behavior in Redisson when using `EVAL`.

This seems to be supported in Spring [spring-data-redis](https://docs.spring.io/spring-data/redis/docs/2.1.0.RC1/reference/html/#scripting) using the default `ScriptExecutor`.

Is there any script caching option I am not aware of? Is this going to be implemented if missing?

**Comment by mrniko** [comment]
Config.useScriptCache settings added. Please refer to wiki for more details https://github.com/redisson/redisson/wiki/2.-Configuration#usescriptcache

**Comment by alturkovic** [comment]
Awesome, thanks for the quick changes!

**Comment by mrniko** [comment]
@alturkovic 

Thank you! Redisson architecture affords that ;)

**Comment by jackygurui** [comment]
Awesome work @mrniko

**Comment by mrniko** [comment]
@jackygurui 

Thank you!

**Comment by andreaugustoaragao** [comment]
if this has better performance, why isn't it true by default? Any gotchas?

**Comment by henry701** [comment]
Yeah, I'm also wondering why it isn't true by default? Considering the widespread use of Lua scripting to power the Redisson abstractions, it could speed up systems by a LOT.
